packages/abap-deploy-config-inquirer/src/prompts/questions/abap-target.ts

```diff
@@ -0,0 +1,261 @@
+            },
+            name: abapDeployConfigInternalPromptNames.targetSystemCliSetter
+        } as Question);
+    }
+    return prompts;
+}
+
+/**
+ * Returns the url prompt.
+ *
+ * @param destinations - list of destinations
+ * @param backendTarget - backend target
+ * @returns input question for url
+ */
+function getUrlPrompt(
+    destinations?: Destinations,
+    backendTarget?: BackendTarget
+): Question<AbapDeployConfigAnswersInternal> {
+    return {
+        when: (previousAnswers: AbapDeployConfigAnswersInternal) => {
+            const isValid = showUrlQuestion(previousAnswers.targetSystem);
+            updatePromptStateUrl(previousAnswers, destinations, backendTarget);
+            return isValid;
+        },
+        type: 'input',
+        name: abapDeployConfigInternalPromptNames.url,
+        message: t('prompts.target.url.message'),
+        guiOptions: {
+            mandatory: true,
+            breadcrumb: true
+        },
+        default: ({ targetSystem }: AbapDeployConfigAnswersInternal): string | undefined => defaultUrl(targetSystem),
+        filter: (input: string): string => input?.trim(),
+        validate: (url: string): boolean | string => validateUrl(url)
+    } as InputQuestion<AbapDeployConfigAnswersInternal>;
+}
+
+/**
+ * Returns the scp prompt.
+ *
+ * @param backendTarget - backend target
+ * @returns confirm question for scp
+ */
+function getScpPrompt(backendTarget?: BackendTarget): Question<AbapDeployConfigAnswersInternal> {
+    return {
+        when: (previousAnswers: AbapDeployConfigAnswersInternal): boolean => showScpQuestion(previousAnswers),
+        type: 'confirm',
+        name: abapDeployConfigInternalPromptNames.scp,
+        message: t('prompts.target.scp.message'),
+        guiOptions: {
+            breadcrumb: t('prompts.target.scp.breadcrumb')
+        },
+        default: (): boolean | undefined => backendTarget?.abapTarget?.scp,
+        validate: (scp: boolean): boolean | string => validateScpQuestion(scp)
+    } as ConfirmQuestion<AbapDeployConfigAnswersInternal>;
+}
+
+/**
+ * Returns the client choice prompt.
+ *
+ * @param backendTarget - backend target
+ * @returns list question for client choice
+ */
+function getClientChoicePrompt(
+    backendTarget?: BackendTarget
+): (YUIQuestion<AbapDeployConfigAnswersInternal> | Question)[] {
+    const prompts: (ListQuestion<AbapDeployConfigAnswersInternal> | Question)[] = [
+        {
+            when: (): boolean =>
+                showClientChoiceQuestion(backendTarget?.abapTarget?.client, PromptState.abapDeployConfig?.isS4HC),
+            type: 'list',
+            name: abapDeployConfigInternalPromptNames.clientChoice,
+            message: t('prompts.target.clientChoice.message'),
+            guiOptions: {
+                applyDefaultWhenDirty: true
+            },
+            choices: () => getClientChoicePromptChoices(backendTarget?.abapTarget?.client),
+            default: (): string => ClientChoiceValue.Blank,
```

Comment:
Should this use the backendTarget?.abapTarget?.client if present otherwise use `Blank`?

Reply:
Yes you're right, I'll update